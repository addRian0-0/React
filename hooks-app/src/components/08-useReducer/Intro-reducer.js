//Nomas lo vi jejeje pero no lo hice, pero si le entendi