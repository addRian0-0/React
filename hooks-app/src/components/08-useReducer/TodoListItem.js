import React from 'react'

export const TodoListItem = ({ todo, handleDelete, handleTodo, index }) => {
    return (
        <div>
            <li
                key={todo.id}
                className="list-group-item"
            >
                <p
                    className={`${todo.done && 'complete'}`}
                    onClick={() => handleTodo(todo.id)}
                >
                    {index + 1}. {todo.desc}
                </p>
                <button
                    className="btn btn-danger"
                    onClick={() => { handleDelete(todo.id) }}
                >
                    Borrar
                </button>
            </li>
        </div>
    )
}
