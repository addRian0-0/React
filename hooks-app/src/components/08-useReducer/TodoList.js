import React from 'react'
import { TodoListItem } from './TodoListItem'

export const TodoList = ({ todos, handleDelete, handleTodo }) => {
    return (
        <div>
            <ul className="list-group list-group-flush">
                {
                    todos.map((todo, i) => {
                        return <TodoListItem
                            todo={todo}
                            key={todo.id}
                            handleDelete={handleDelete}
                            handleTodo={handleTodo}
                            index={i}
                        />
                    })
                }
            </ul>
        </div>
    )
}
