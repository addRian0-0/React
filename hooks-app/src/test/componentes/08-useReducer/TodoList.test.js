import { shallow } from "enzyme";
import { TodoList } from "../../../components/08-useReducer/TodoList";
import { demosTodo } from "../../fixtures/demoTodos";

describe('pruebas en el todo list', () => {

    const handleDelete = jest.fn()
    const handleTodo = jest.fn()

    const wrapper = shallow(
        <TodoList
            todos={demosTodo}
            handleDelete={handleDelete}
            handleTodo={handleTodo}
        />
    )

    test('debe de mostrarse correctamente ', () => {

        expect(wrapper).toMatchSnapshot();

    })

    test('debe de tener dos todosListItem ', () => {

        expect(wrapper.find('TodoListItem').length).toBe(demosTodo.length);
        expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function));

    })


})
