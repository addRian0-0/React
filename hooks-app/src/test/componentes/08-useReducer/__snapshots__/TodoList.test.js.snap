// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el todo list debe de mostrarse correctamente  1`] = `
<div>
  <ul
    className="list-group list-group-flush"
  >
    <TodoListItem
      handleDelete={[MockFunction]}
      handleTodo={[MockFunction]}
      index={0}
      key="1"
      todo={
        Object {
          "desc": "Aprender fastify",
          "donde": false,
          "id": 1,
        }
      }
    />
    <TodoListItem
      handleDelete={[MockFunction]}
      handleTodo={[MockFunction]}
      index={1}
      key="2"
      todo={
        Object {
          "desc": "Aprender typescript",
          "donde": false,
          "id": 2,
        }
      }
    />
  </ul>
</div>
`;
